src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java

```diff
@@ -126,6 +136,12 @@ private MessageParams msg(final MessageType sev, final String cont) {
         return Files.readAllLines(f.toPath());
     }
 
+    private File designatedTemporaryFile(File source) {
+        String hashed = hash.hashString(source.getAbsolutePath(), StandardCharsets.UTF_8).toString();
```

Comment:
I think this is better than constantly creating and cleaning up temporary files.

Easier to work with stable paths to provide accurate diagnostics on temporary files.